infra/templates/az-isolated-service-single-region/app.tf

```diff
@@ -33,3 +33,76 @@ module "keyvault" {
     "azurerm" = "azurerm.app_dev"
   }
 }
+
+module "container_registry" {
+  source                  = "../../modules/providers/azure/container-registry"
+  container_registry_name = local.acr_name
+  resource_group_name     = azurerm_resource_group.app_rg.name
+  // Note: this is requird until App Services and ACR work over MSI. See the design document for more details.
+  container_registry_admin_enabled = true
+  // Note: only premium ACRs allow configuration of network access restrictions
+  container_registry_sku = "Premium"
+  providers = {
+    "azurerm" = "azurerm.app_dev"
+  }
+}
+
+# Build and push the app service image slot to enable continuous deployment scenarios.
+# We're using ACR build tasks to remotely carry out the docker build / push.
+resource "null_resource" "acr_image_deploy" {
+  count      = length(var.deployment_targets)
+  depends_on = [module.container_registry]
+
+  triggers = var.deployment_targets[count.index]
+
+  provisioner "local-exec" {
+    command = "az acr build -t $IMAGE_TAG -r $DOCKER_REPOSITORY -f $DOCKER_FILE $GIT_SOURCE"
+
+    // Note: this is needed to prevent failure in the case that the following order of operations occurrs:
+    //  (1) Initial template deployment is configured to use N webapps. This triggers (in sequence) N ACR
+    //      build tasks that seed the ACR with the initial container images. Then the ACR network rules
+    //      are constrained to only allow access from within the configured VNET.
+    //  (2) A subsequent deployment specifies N + 1 webapps. This will attempt to trigger an ACR build task
+    //      that builds and pushes the new container to the ACR. However, this will fail due to the network
+    //      restrictions configured after the initial ACR build tasks ran.
+    //
+    // This flag will ignore the failed build from (2). The implication is that the webapp will deploy
+    // successfully. Hitting the HTTP(s) endpoint will return an error response because the container that
+    // should be deployed does not exist. The fix for this is to build and push the image to the ACR from
+    // within the VNET, which will be done as a part of the app dev CI/CD process.
+    on_failure = "continue"
```

Comment:
It's unclear how we'd bubble up errors if the initial docker build step were to fail?

Reply:
If the initial step fails then the integration tests should catch the error.